pkg/image/apis/image/helper.go

```diff
@@ -436,6 +469,9 @@ func ImageWithMetadata(image *Image) error {
 
 	if len(image.DockerImageLayers) > 0 && image.DockerImageMetadata.Size > 0 && len(image.DockerImageManifestMediaType) > 0 {
 		glog.V(5).Infof("Image metadata already filled for %s", image.Name)
+
+		ReorderImageLayers(image)
```

Comment:
Shouldn't we update the image object in etcd if there's such a metadata change?
That way, we could allow for easy mass migration if desired (let the user get all the images).

Reply:
The `ImageWithMetadata` used in the imageAPI ([for create](https://github.com/openshift/origin/blob/master/pkg/image/registry/image/strategy.go#L42), [for update](https://github.com/openshift/origin/blob/master/pkg/image/registry/image/strategy.go#L115)), in the importer etc. So, as the images change, they will be converted.

Reply:
@miminar I don't want to make extra requests to database.